### PR TITLE
LOOP-1321 LOOP-1323 Isf and correction ranges updates for swift UI and guardrails

### PR DESCRIPTION
**Pull request by jh-bate**

### CODE REVIEW

File: e2e/functional/functional_test.spec.js [hunk 1]
```diff
@@ -24,16 +24,12 @@ describe('functional test', () => {
...
-        //TODO: update when development work complete
-        describe.skip('correction range schedule', () => {
+        describe('correction range schedule', () => {
             guardrailsTests.correctionRangeScheduleTests(test);
         });
         //TODO: update when development work complete
         describe.skip('insulin carb ratio', () => {
             guardrailsTests.insulinCarbRatioTests(test);
         });
-        describe.skip('insulin sensitivity schedule', () => {
```
Comment (darinkrauss) [inline e2e/functional/functional_test.spec.js:35]: [nit] No longer having these tests?

File: src/subScreen/basalRatesScreen.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        return match.accessible.ButtonBarButton(this.language.general.Add);
+    }
+    async Save() {
+        await this.SaveButton().tap();
+    }
+    async Close() {
+        await this.BackButton().tap();
+    }
+    /**
+     * @summary basal rate to be set. NOTE: it is assumed that the rates are given in order of time
```
Comment (darinkrauss) [inline src/subScreen/basalRatesScreen.js:28]: [nit] Comment not necessary for applying just one?

File: src/subScreen/basalRatesScreen.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    * @param {Object} rate
+     * @param {String} rate.time
+     * @param {String} rate.unitsPerHour
+     */
+    async Apply(rate) {
```
Comment (darinkrauss) [inline src/subScreen/basalRatesScreen.js:33]: [nit] Consider `Apply` -> `ApplyOne` to be completely clear?

File: src/subScreen/basalRatesScreen.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        if (rate) {
+            await this.AddButton().tap();
+            await match.accessible.Label(`${rate.time}`).atIndex(0).tap();
+            await action.SetPickerValue(1, `${rate.unitsPerHour} ${config.basalRatesUnits}`);
+        }
+    }
+    /**
+     * @summary basal rates to be set. NOTE: it is assumed that the rates are given in order of time
```
Comment (darinkrauss) [inline src/subScreen/basalRatesScreen.js:41]: [nit] Consider sorting `rates` on `time` to validate? Also does this imply that `rates[0].time` must also be `0`?

File: src/subScreen/insulinSensitivitiesScreen.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        });
+        this.config = config;
+    }
+    /**
+     * @param {Object} expectedSensitivity
+     * @param {String} expectedSensitivity.time
+     * @param {String} expectedSensitivity.bgValuePerInsulinUnit
+     * @param {Object} currentSensitivity optional
+     */
+    async Apply(expectedSensitivity, currentSensitivity) {
```
Comment (darinkrauss) [inline src/subScreen/insulinSensitivitiesScreen.js:21]: [nit] What is the purpose of the `currentSensitivity`?

File: tests/accessibility/bolus_screen.tests.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+var bolusScreenAccessibilityTests = (test) => {
```
Comment (darinkrauss) [inline tests/accessibility/bolus_screen.tests.js:1]: [nit] I notice some filenames use camelcase and others use snakecase?

File: tests/guardrails/correction_range_schedule.tests.js [hunk 7]
```diff
@@ -2,49 +2,90 @@ const { limits } = require('../../src/index');
...
-    it('can set at max', async () => {
+    it('can set max units at limit', async () => {
+        await screen.Add();
         await screen.Apply({
-            min: limits.correctionRange.max.limit,
+            min: 100,
             max: limits.correctionRange.max.limit,
         });
+        await screen.AddNewEntry();
+        await expect(screen.GuardrailWarningIconPicker(0)).toBeVisible();
```
Comment (darinkrauss) [inline tests/guardrails/correction_range_schedule.tests.js:18]: [nit] What is the number parameter on `GuardrailWarningIconPicker`?

**Review by darinkrauss — APPROVED** [review]
LGTM. Some nits and a couple questions, but nothing major. 🦙

File: e2e/functional/functional_test.spec.js (hunk 1 above)
Comment (jh-bate) [reply e2e/functional/functional_test.spec.js:35]: its still in the file but is no longer skipped 

```
describe('insulin sensitivity schedule', () => {
      guardrailsTests.insulinSensitivityScheduleTests(test);
});
```

File: src/subScreen/basalRatesScreen.js (hunk 3 above)
Comment (jh-bate) [reply src/subScreen/basalRatesScreen.js:33]: yeah - i might roll that change in next PR

File: tests/accessibility/bolus_screen.tests.js (hunk 6 above)
Comment (jh-bate) [reply tests/accessibility/bolus_screen.tests.js:1]: yeah good point - I will convert to one style

File: tests/guardrails/correction_range_schedule.tests.js (hunk 7 above)
Comment (jh-bate) [reply tests/guardrails/correction_range_schedule.tests.js:18]: yeah maybe a named param would help

**Pull request merged**